workspaces/ai-integrations/plugins/ai-experience/src/components/NewsPage/NewsGrid.tsx

```diff
@@ -0,0 +1,153 @@
+  const theme = useTheme();
+  const rssApi = useApi(rssApiRef);
+  const [articleData, setArticleData] = useState<Article[]>([]);
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    const fetchRSS = async () => {
+      try {
+        const response = await rssApi.fetch();
+
+        const sanitizedText = sanitizeXML(response);
+
+        const result = await parseStringPromise(sanitizedText, {
+          trim: true,
+          explicitArray: false,
+          mergeAttrs: true,
+          tagNameProcessors: [processors.stripPrefix],
+        });
+
+        const items = result.rss?.channel?.item || [];
+        const parsedItems = (Array.isArray(items) ? items : [items]).map(
+          (item: any) => {
+            const thumbnail =
+              item.thumbnail?.url ||
+              item.content?.url ||
+              item['media:thumbnail']?.url ||
+              item['media:content']?.url ||
+              item['itunes:image']?.href ||
+              extractImageFromHTML(item.description);
+
+            return {
+              title: item.title,
+              link: item.link,
+              pubDate: item.pubDate,
+              description: item.description,
+              thumbnail,
+            };
+          },
+        );
+
+        setArticleData(parsedItems);
+      } catch (error) {
+        // eslint-disable-next-line no-console
+        console.error('Error fetching RSS feed:', error);
+      } finally {
+        setLoading(false);
+      }
+    };
+    fetchRSS();
+  }, [rssApi]);
+
+  if (loading) {
+    return (
+      <Box
+        sx={{
+          display: 'flex',
+          flexDirection: 'column',
+          alignItems: 'center',
+          justifyContent: 'center',
+          width: '100%',
+          padding: '40px 0',
+          backgroundColor: theme.palette.background.paper,
+        }}
+      >
+        <CircularProgress />
+        <Box sx={{ mt: 2 }}>Fetching RSS Feed</Box>
+      </Box>
+    );
+  }
+
+  if (!articleData || articleData.length === 0) {
+    return (
+      <Box
+        sx={{
+          display: 'flex',
+          flexDirection: 'column',
+          alignItems: 'center',
+          justifyContent: 'center',
+          width: '100%',
+          padding: '40px 0',
+          backgroundColor: theme.palette.background.paper,
+        }}
+      >
+        <Box sx={{ textAlign: 'center', color: theme.palette.text.secondary }}>
+          <Box sx={{ fontWeight: 'medium', fontSize: '1rem', mb: 1 }}>
+            No RSS Content Available
+          </Box>
+          <Box sx={{ fontSize: '0.875rem' }}>
+            Unable to retrieve content from the specified RSS source. Please
+            verify the RSS feed URL or try an alternative source.
+          </Box>
+        </Box>
+      </Box>
+    );
+  }
+
+  return (
+    <Box
+      sx={{
+        alignItems: 'center',
+        justifyContent: 'space-between',
+        padding: '36px 240px 48px 40px',
```

Comment:
nit: It is better to use theme.spacing function rather then setting px values directly, so that if theme spacing globally in the future, then the layout would react automatically.

```suggestion
        padding: `${theme.spacing(4.5)} ${theme.spacing(30)} ${theme.spacing(
          6,
        )} ${theme.spacing(5)}`,
```